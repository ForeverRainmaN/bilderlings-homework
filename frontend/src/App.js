import React, {Component} from 'react';
import {BrowserRouter, Route, Switch} from "react-router-dom";
import {FeeEditor} from "./components/FeeEditor/FeeEditor";
import CurrencyCalculator
  from "./components/CurrencyCalculator/CurrencyCalculator";
import '../node_modules/normalize.css/normalize.css';
import '../node_modules/@blueprintjs/core/lib/css/blueprint.css';
import './App.css';

class App extends Component {
  render() {
    return (
        <div className="App">
          <BrowserRouter>
            <Switch>
              <Route path="/" exact component={CurrencyCalculator}/>
              <Route path="/fees" component={FeeEditor}/>
            </Switch>
          </BrowserRouter>
        </div>
    );
  }
}

export default App;
