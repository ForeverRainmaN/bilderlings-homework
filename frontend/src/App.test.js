import React from "react";
import App from "./App";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";

it('tests if App renders correctly', () => {
  const addFeeForm = renderer
  .create(
      <MemoryRouter>
        <App/>
      </MemoryRouter>).toJSON();
  expect(addFeeForm).toMatchSnapshot();
});
