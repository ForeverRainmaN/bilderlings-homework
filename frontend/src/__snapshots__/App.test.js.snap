// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests if App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="bilderlings-homework-curr-calc display-flex"
  >
    <a
      href="/fees"
      onClick={[Function]}
    >
      Fee Editor
    </a>
    <div
      className="display-flex"
    >
      <div
        className="bilderlings-homework-amount-input width70-br5-mr-30"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="off"
            className="bp3-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
            value="0"
          />
        </div>
      </div>
      <div
        className="bilderlings-homework-add-fee-form-selects display-flex"
      >
        <div
          className="bp3-select"
        >
          <select
            id="from"
            onChange={[Function]}
          />
        </div>
        <div
          className="bp3-select"
        >
          <select
            id="to"
            onChange={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="bilderlings-homework-converted-amount"
    >
      <h1>
        0.00
      </h1>
    </div>
  </div>
</div>
`;
